pkg/reconciler/route/resources/ingress.go

```diff
@@ -96,9 +98,10 @@ func MakeIngressSpec(
 	}
 	// Sort the names to give things a deterministic ordering.
 	sort.Strings(names)
-
 	// The routes are matching rule based on domain name to traffic split targets.
-	rules := make([]v1alpha1.IngressRule, 0, len(names))
+	rules := make([]v1alpha1.IngressRule, 0, len(names)+len(acmeChallenges))
+	rules = append(rules, makeACMEIngressRules(acmeChallenges)...)
```

Comment:
I tried this code using gloo (https://github.com/solo-io/gloo)
and I get the following error when I do:

```
kubectl describe proxy -n gloo-system knative-external-proxy
```
I get:
```
VirtualHost Error: DomainsNotUniqueError. Reason:
    domain <myapp>.<mynamespace>.<mydomain>.com is shared by the following virtual
    hosts: [<mynamespace>.<myapp>-0 <mynamespace>.<myapp>-1];
```
(Replace myapp, mynamespace and mydomain with yours)

So it seems that this is compatible with istio but not with gloo. Anyone else has tried it?

Reply:
/cc @ilackarms 
We leverage the path based routing of Knative Ingress (https://github.com/knative/serving/blob/master/pkg/apis/networking/v1alpha1/ingress_types.go#L209) in order to route the traffic for ACME HTTP01 challenge validation to a (cert-manager) service for fulfilling the challenge.
I am not sure how gloo implements it. Would like to get more insights from glood. :)

Reply:
the issue is occurring because there are now multiple rules in the `ingress` which share a hostname. Gloo works under the assumption currently that these will be unique. istio merges conflicting domains while Gloo errors on them. is this assumption not correct? if so, Gloo will require a patch to merge ingress rules by hostname.

Reply:
Thanks @ilackarms for the insights!

It sounds reasonable to me to assume that domains across all IngressRules will be unique. We may want to document it.

Specific for the implementation here, we can append the routing rule for ACME validation traffic into the [`Path`](https://github.com/knative/serving/blob/master/pkg/apis/networking/v1alpha1/ingress_types.go#L192) of an existing `IngressRule` instead of having multiple `IngressRule`.